challenge_4/c/scottsquatch/src/binary_tree.h

```diff
@@ -0,0 +1,130 @@
+// Minimal BinaryTree. Only implement methods necessary for solving
+// problem 3.
+#ifndef _BINARY_TREE_H_
+#define _BINARY_TREE_H_
+
+#include <stdio.h>
+#include <stdlib.h>
+
+/* Structs */
+struct Node
+{
+  int data;
+  struct Node* left;
+  struct Node* right;
+};
+
+struct BinaryTree
```

Comment:
You can use `typedef` to give these structures a type name. This would eliminate having to type struct so often below.

`typedef struct { struct Node* root; } BinaryTree;`

And from then on you can just refer to `BinaryTree`.